### PR TITLE
Add `ifEqual` type guard

### PR DESCRIPTION
**Pull request by harsh-srivastava44**
Added new type IfEqual type guard, to improve on the issue raised #788

### CODE REVIEW

**Comment by harsh-srivastava44** [comment]
@sindresorhus  @Emiyaaaaa can you guys please review this ?

**Comment by Emiyaaaaa** [comment]
Could you add this type to README.md?

**Comment by sindresorhus** [comment]
Make sure you read and adhere to: https://github.com/sindresorhus/type-fest/blob/main/.github/contributing.md#submitting-a-new-type

**Comment by harsh-srivastava44** [comment]
@Emiyaaaaa this is updated and added doc for all available `IfT` utilities

**Comment by harsh-srivastava44** [comment]
@sindresorhus is it good to be merged ?

**Comment by sindresorhus** [comment]
https://github.com/sindresorhus/type-fest/pull/792#issuecomment-1880209587

Go through each point and make sure you follow it.

**Comment by harsh-srivastava44** [comment]
@sindresorhus I have just started with contribution journey , It would be helpful to let me know what missing in the PR?

File: readme.md [hunk 1]
```diff
@@ -217,6 +217,13 @@ type ShouldBeNever = IfAny<'not any', 'not never', 'never'>;
 - [`IsNever`](source/is-never.d.ts) - Returns a boolean for whether the given type is `never`. (Conditional version: [`IfNever`](source/if-never.d.ts).)
 - [`IsUnknown`](source/is-unknown.d.ts) - Returns a boolean for whether the given type is `unknown`. (Conditional version: [`IfUnknown`](source/if-unknown.d.ts).)
 
+Currently, `IfType` guards are available for the following checks:
+
+-   [`IfAny`](source/if-any.d.ts) - Determines if a type is `any`, and returns `TypeIfAny` if true, otherwise `TypeIfNotAny`.
```
Comment (sindresorhus) [inline readme.md:222]: I think it's enough to do this so we don't have to duplicate the description:
```suggestion
-   [`IfAny`](source/if-any.d.ts) - Conditional version of [`IsAny`](source/is-any.d.ts).
```

File: source/if-equal.d.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import type {IsEqual} from './is-equal';
+
+/**
+ * An if-else-like type that resolves depending on whether the two given types are equal.
+ *
+ * @see {@link IsEqual}
+ *
```
Comment (sindresorhus) [inline source/if-equal.d.ts:7]: Doc comments should not be prefixed by `*`

**Comment by sindresorhus** [comment]
> I have just started with contribution journey , It would be helpful to let me know what missing in the PR?

Reviewing PRs is a lot of work. That's why we have the contribution guidelines. It points out common issues so you can fix it yourself before taking a maintainers time to point out obvious things. Just go through it point by point.

**Pull request closed without merging**